eclipse-sw360/sw360
# feat(rest): new endpoint `/moderationrequest`

[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

The PR adds new endpoint `/moderationrequest` to allow listing ModerationRequests from REST API.

* Which issue is this pull request belonging to and how is it solving it?
> PR is related to issue #1849
* Did you add or update any new dependencies that are required for your change?
> No

Issue: #1849

### Suggest Reviewer
> You can suggest reviewers here with an @mention.

### How To Test?
> How should these changes be tested by the reviewer?

Reviewer can try to load the following implemented endpoints:
- `GET` `/moderationrequest`
  - Should respond with all moderation requests available to the user.
  - The endpoint should be paginated.
- `GET` `/moderationrequest/{id}`
  - Should respond with single moderation request.
  - Additionally, should embed the requesting user's object and relevant project/component/release/license.
- `GET` `/moderationrequest/byState?state={state}`
  - Should filter only moderation requests with given state.
  - State should be checked against possible values of `open` & `closed`. Other values should throw 400 bad request.
> Have you implemented any additional tests?

No

### Checklist
Must:
